docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,132 @@
+
+### Save user provided explicit flags
+The user should not need to explicitly save the configuration in this file, but the odo CLI should keep the values explicitly used by a user into this file.
+
+- If the file does not exist, it should create it.
+- If the value is already present in the file, it should overwrite it.
+
+```
+odo create wildfly example --memory=1GB
+
+# This should set value in .odo config file   (memory=2GB)
+```
+
+### Use configuration to "decorate" the deployment
+On component creation, via `odo create ...` the CLI will take information stored in this configuration file to create an actual deployment on OpenShift.
+
+```
+# Value in .odo config file   (memory=2GB)
+odo create wildfly example
+```
+
+The deployment that should be created, would have the following configuration:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+spec:
+  containers:
+  - image: xxx
+    resources:
+      limits:
+        memory: 2Gi
+      requests:
+        memory: 2Gi
+```
+
+### Use configuration to "decorate" the CLI behavior
+On component creation, via `odo create ...` the CLI will take information stored in this configuration file to define some values that would typically be defined by the CLI and that are relative to how this works..
+
+```
+# Value in .odo config file   (app-name=myapp, component-name=mycomp)
+odo create wildfly
+# Would create an application called myapp and component named mycomp using wildfly
+```
+
+
+### Developer friendly flags
+Configuration options that will decorate the deployment should try to not use (as much as possible) terminology used in k8s. An example are memory resource requests and limits. Instead one could use terms like max-mem, min-mem and mem, that could work like this:
+
+* When `mem` is set, memory requests and limits are the same.
+* `min-mem` should be memory requests.
+* `max-mem` should be memory limits. 
+* `mem` is exclusive with `min-mem` or `max-mem`
+
+### Configuration precedence
+If the CLI provides some configuration values as flags for configuration that is store in the file, the ones provided explicitly from the user should take precedence.
+
+```
+# Value in .odo config file   (memory=2GB)
+odo create wildfly example --memory=1GB
+
+# The value that should be set for memory is 1 GB
+```
+
+### Explicitly modify the .odo config file
+There might be times that a developer would want to add/edit/remove values from the .odo configuration file. There should be a command to manage this file in the odo CLI. That command should be the same used to touch configuration of the global `.odo` config file located in the users's home directory.
+
+Currently that command is:
+
+```
+odo utils config [set|get|delete|add] KEY [VALUE] --local
+```
+
+Some examples of this command could be:
+
+```
+odo utils config set memory 2Gi --local      # Set's component memory to 2 GB
```

Comment:
this `--local` flag is a bit confusing?
can we change it to something like global ? 
If you mean `--local` by set the memory value on the local file lies in `$HOME/.kube/odo`

Reply:
got it from @kadel 's previous comment